src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -892,9 +923,26 @@ def _get_hms_fed_tables(self, tables: list[Table], target_catalog) -> list[Table
     def _migrate_acls(self, tables_in_scope: list[TableToMigrate]) -> None:
         tasks = []
         for table in tables_in_scope:
-            tasks.append(partial(self._migrate_grants.apply, table.src, table.rule.as_uc_table_key))
+            tasks.append(partial(self._migrate_grants.apply, table.src, table.rule.as_uc_table))
         Threads.strict("migrate grants", tasks)
 
+    def _retrieve_acls(self, tables_in_scope: list[TableToMigrate]) -> Iterable[Grant]:
+        tasks = []
+        for table in tables_in_scope:
+            tasks.append(partial(self._migrate_grants.retrieve, table.src, table.rule.as_uc_table))
+        grants, errors = Threads.gather("retrieve grants", tasks)
```

Comment:
we don't make any network calls in `retrieve` (after caching), so adding multithreading here increases complexity without any added benefit. rewrite this as synchronous implmentation

Reply:
KK